form-design.rst

```diff
@@ -0,0 +1,420 @@
+
+Our experience is that these form design tools can provide a good starting point, but, to enable advanced features, you will inevitably need to edit the resulting form. After hand-editing, you should verify the syntax of the form using ODK Validate.
+
+.. xform-file:
+
+Overview of an XForm file
+==========================
+
+XForms consist of four major components:
+
+- The **title** -- the name of the form. This will be shown to the user.
+- The **instance** -- where your data are going to be stored. For each prompt, you need a place to store your answer.
+- The **bindings** -- for each instance variable, you can include information about those data like the type (string, integer, etc.), the constraints on those data (i.e. less than 3), whether they're required or not, etc.
+- The **body** -- how prompts are displayed to the user.
+
+These are explained in the following HTML example one by one.
+
+.. code-block:: html
+
+  <h:html xmlns="http://www.w3.org/2002/xforms"
+  xmlns:h="http://www.w3.org/1999/xhtml"
+  xmlns:ev="http://www.w3.org/2001/xml-events"
+  xmlns:xsd="http://www.w3.org/2001/XMLSchema"
+  xmlns:jr="http://openrosa.org/javarosa">
+
+  <h:head>
+  <h:title>Simple Form</h:title>
+  <model>
+
+  <!-- instance -->
+  <instance>
+   <data id="myform" >
+     <meta>
+       <instanceID/>
+     </meta>
+     <mystring/>
+   </data>
+  </instance>
+
+  <!-- bindings -->
+  <bind nodeset="/data/meta/instanceID" type="string" 
+           readonly="true()" calculate="concat('uuid:',uuid())" />
+  <bind nodeset="/data/mystring" type="string"/>
+
+  </model>
+  </h:head>
+
+  <!-- body -->
+  <h:body>
+  <input ref="/data/mystring">
+    <label>This is a string prompt</label>
+    <hint>You don't need a hint, but you can add one</hint>
+  </input>
+  </h:body>
+
+  </h:html>
+
+Instances
+----------
+
+.. code-block:: html
+
+  <instance>
+   <data id="myform" >
+    <meta>
+      <instanceID/>
+    </meta>
+    <mystring/>
+  </data>
+  </instance>
+
+Here we have a single instance called "data". This instance is identified via the 'id' attribute as the 'myform' form. This id attribute, and an optional version attribute, provide the internal unique name of this form to the software. The software only allows one form definition for each (id, version) pair.
```

Comment:
> we have a single instance called "data"

I know this just came from the original documentation, but I don't know that this is sensible way to talk about this.

I'm opening a discussion issue because I think this language (Which is all over the old docs) is confusing. It is also not exactly consistent with what is going on in the ODK XLSForm output.